### PR TITLE
feat(airbyte-ci): add deadlineAction to metadata

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/airbyte-internal-issues/issues/10081

More context here: https://docs.google.com/document/d/1ceId9f2pCAg6llGDNRqlHrUnaNpmKlcwF6f0Qd5_OYk/edit#heading=h.glibu2q3ptc1

## How

add new field  `deadlineAction` to metadata

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorBreakingChanges.py [hunk 1]
```diff
@@ -27,6 +28,9 @@ class Config:
 
     upgradeDeadline: date = Field(..., description="The deadline by which to upgrade before the breaking change takes effect.")
     message: str = Field(..., description="Descriptive message detailing the breaking change.")
+    deadlineAction: Optional[Literal["auto_upgrade", "disable"]] = Field(
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorBreakingChanges.py:31]: Nice 👏🏼

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml [hunk 2]
```diff
@@ -25,6 +25,13 @@ definitions:
       message:
         description: Descriptive message detailing the breaking change.
         type: string
+      deadlineAction:
```
Comment (natikgadzhi) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml:28]: This having a default value means that we don't _have_ to backfill the existing connectors, right?

**Comment by natikgadzhi** [comment]
@alafanechere give this a review please.

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml (hunk 2 above)
Comment (natikgadzhi) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml:28]: @alafanechere do we want to backfill, or nah? 

If we don't backfill, do we want a followup PR with metadata schema docs update? :D

**Review by natikgadzhi — APPROVED** [review]

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml (hunk 2 above)
Comment (alafanechere) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml:28]: Good question.
As registry entries are generated on publish, we would have to backfill already published metadata / registry entries for this kind of change if we want to expose the default value in the registry.

IMO the default value should be handled platform side as it's a global behavior that could change between OSS and Cloud.
When we'll work on consuming this `deadlineAction` in the platform we can make it set `auto_upgrade` when `deadlineAction` is null.

**So my recommendation is to  set a default value and keep this field optional.**
* No backfill will be needed
* Platform will use the metadata defined `deadlineAction` if it's declared or fallback to its default deadlineAction (like `auto_upgrade`).

Wdyt?

**Review by alafanechere — CHANGES_REQUESTED** [review]
Requesting change as I think the default `deadlineAction` should be set on the platform side. This would avoid backfilling all existing metadata and registry entries and sounds righter to me.

Can you please bump the `lib` package version and run `poetry lock` in the `orchestrator` package? This will make the orchestrator pick up the updated models.

**Comment by artem1205** [comment]
Need to update python version in https://github.com/airbytehq/airbyte/pull/46312/commits/cb33fb85959cddc538d68720683fd36692b684c6#diff-481077d02a897b929582f9f8c9f8ac73d036a963915b1c83dff68a7c93875335R10

**Comment by artem1205** [comment]
Hey @alafanechere , could you please clarify why is it impossible to leave the old version (3.9) or how to fix imports in the tests? Am I missing something?

**Comment by alafanechere** [comment]
> Hey @alafanechere , could you please clarify why is it impossible to leave the old version (3.9) or how to fix imports in the tests? Am I missing something?

@artem1205 can you update your branch? I fixed that this morning I think.

**Comment by artem1205** [comment]
@alafanechere , is there anything else that should be done here?

**Comment by alafanechere** [comment]
> @alafanechere , is there anything else that should be done here?

Yes, please [checkout my comment](https://github.com/airbytehq/airbyte/pull/46312/files#r1795018744)

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml (hunk 2 above)
Comment (artem1205) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/ConnectorBreakingChanges.yaml:28]: I'm okay with removing the default value if it's going to be set on the platform.

**Comment by artem1205** [comment]
> Yes, please [checkout my comment](https://github.com/airbytehq/airbyte/pull/46312/files#r1795018744)

Done!

**Comment by artem1205** [comment]
@alafanechere , any updates on this?

**Review by alafanechere — APPROVED** [review]
Thanks for the changes @artem1205

**Pull request merged**